W12-Revision.md

```diff
@@ -795,149 +795,137 @@
-          - Sub class should be able to substitute the super class
 
-        - Interface Segregation Principle (ISP)
-          - Client should not be forced to depend on method that they do not use.
+<details>
+	<summary> Show answer </summary>	
+
+	Single Responsibility Princple - Each component in our system should only have one purpose. Giving a component
+					too many tasks allows for the system to be more prone to have bugs.
+					
+	Open Closed Princple - Open for extension, but closed for modifications. When we want to extend our system, 
+			       we shoudn't have to modify too much of our existing code.
+	
+	Liskov Substitution Princple - Sub classes should be able to substitute super classes. This means that 
+				       if we were to make a test and we want test the speed of a "truck", then we should 
+				       also be able to substitute the "truck" class with "vehicle" instead.
+				
+	Interface Segregation Princple - Clients should not be forced to depend on the method that they do not use.
+				         For example if we have one class called "Person", if we had a "Kid" class, 
+					 then the "Kid" class should not be able to access code from the "Adult' class.
+					 Hence, it would be beneficial to make the Person class an abstract class and 
+					 we should give the appropriate methods for each of the classes.
+					 
+	Dependency Inversion Principle - High level modules should not depend on lower level model. Instead, they 
+					 should depend on their abstractions. For example, a super class method should
+					 not run their subclass methods as they are specific.
+</details>
 
-        - Dependency Inversion Principle (DIP)
-          - High level modules should not depend on low-level modules. 
-          - That's it, abstractions should not depend on details, details should depends on abstraction.
-    </details>
 
-    
 - What is Refactoring, why is it important/critical in Agile. You do not have to memorise the catalog of Refactoring operations, but you should be able to recognise an opportunity for application of a simple Refactoring operation. 
-    <details>
-    <summary>Show answer</summary>
 
-        Refactor is the process of changing a software system in such way that does not alter external behaviour but improves internal structure.
+<details>
+	<summary> Show answer </summary>	
 
-        Refactoring acts as an important factor in order to increase the extensibility and maintenance of a program.
+	Refactoring allows for code management. Refactoring is the process of changing a software system that does not
+	alter the behaviour of the code but improve its internal structure. Agile processes allows for developers to "go back" 
+	and make modifications or extend the code if required. If we "go back" that means that we must look back at our 
+	existing code. Therefore, code should be manageable and easily maintainable if we wish to go back to make 
+	incremental and iterative changes.		   
```

Comment:
Clearer answer for the first part but you didn't answer the second question. What is some simple refactoring operation?